scripts/workspace-conversion/workspace-to-dataset.py

```diff
@@ -0,0 +1,248 @@
+# ------------------------------------------------------------------------------
+# Workspace to Dataset
+#
+# This script converts terra workspaces attributes into a potential catalog entry.
+#
+# Before Running:
+#   python3 -m venv .env
+#   source .env/bin/activate
+#   pip install --upgrade pip
+#   python3 -m pip install requests
+#
+#   Authorization: Access the password in the vault using the appropriate command:
+#      (you will need this for the auth dialog that the script will pull up)
+#   vault login -method=github token=$(cat ~/.github-token)
+#   vault read secret/dsde/terra/kernel/dev/dev/catalog/tests/userAdmin
+#
+# Run:
+#   `python3 workspace-to-dataset.py`
+#   `export GCLOUD_USER={your-email}; export WORKSPACE_NAMESPACE={workspace-namespace}; export WORKSPACE_NAME={workspace-name}; python3 workspace-to-dataset.py`
+#
+# Environment Variables:
+#   GCLOUD_USER: user account email
+#   WORKSPACE_NAMESPACE: workspace namespace for the rawls query
+#   WORKSPACE_NAME: workspace name for the rawls query
+#   RAWLS_URL: url for rawls. Default set to rawls-prod
+# ------------------------------------------------------------------------------
+import json
+import os, subprocess, sys
+import requests
+
+urlRoot = os.environ.get("RAWLS_URL") or "https://rawls.dsde-prod.broadinstitute.org"
+urlWorkspace = f"{urlRoot}/api/workspaces"
+workspaceNamespace = os.environ.get("WORKSPACE_NAMESPACE")
+workspaceName = os.environ.get("WORKSPACE_NAME")
+user = os.environ.get("GCLOUD_USER") or "datacatalogadmin@test.firecloud.org"
+
+policyMap = {
+    "no restrictions": "TerraCore:NoRestriction",
+    "no restriction": "TerraCore:NoRestriction",
+    "general research use": "TerraCore:GeneralResearchUse",
+    "no population origins or ancestry research": "TerraCore:NPOA",
+    "no general methods research": "TerraCore:NMDS",
+    "genetic studies only": "TerraCore:GSO",
+    "clinical care use": "TerraCore:CC",
+    "publication required": "TerraCore:PUB",
+    "collaboration required": "TerraCore:COL",
+    "ethics approval required": "TerraCore:IRB",
+    "geographical restriction": "TerraCore:GS",
+    "publication moratorium": "TerraCore:MOR",
+    "return to database/resource": "TerraCore:RT",
+    "non commercial use only": "TerraCore:NCU",
+    "not-for-profit use only": "TerraCore:NPC",
+    "not for profit use only": "TerraCore:NPC",
+    "not-for-profit, non-commercial use ony": "TerraCore:NPC2",
+}
+
+dataModalityMap = {
```

Comment:
Are there any modalities from here that are relevant? https://github.com/DataBiosphere/terra-data-catalog/blob/main/scripts/hca/create-hca-collection.py#L59

Reply:
oh! I'll merge that map in! I got this list of possible modalities from Kathy - And she recommended filtering out anything that didnt match it. Having the other mapping matches the data from the field better!